sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala

```diff
@@ -111,6 +112,31 @@ private[hive] object HiveShim {
     }
   }
 
+  private def isSubClassOf(t: Type, parent: Class[_]): Boolean = t match {
+    case cls: Class[_] => parent.isAssignableFrom(cls)
+    case _ => false
+  }
+
+  private def hasInheritanceOf[UDFType: ClassTag](funcName: String, clazz: Class[_]): Boolean = {
+    val clsTag = classTag[UDFType].runtimeClass
+    if (isSubClassOf(clazz, clsTag)) {
+      val funcClass = clazz.getMethod(funcName, classOf[MapredContext])
+      funcClass.getDeclaringClass != clsTag
+    } else {
+      false
+    }
+  }
+
+  def validateHiveUserDefinedFunction(udfClass: Class[_]): Unit = {
+    if (hasInheritanceOf[GenericUDF]("configure", udfClass) ||
```

Comment:
I checked it seemed this API had been implemented in v0.11.0 (https://github.com/apache/hive/blob/release-0.11.0/ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDF.java). This is [the commit](https://github.com/apache/hive/commit/a4705e51809e0f6c24cdc417c91600627cbe3482#diff-3111a4cd48c72a32f7713451d7b9ca1aR98) for this API.